src/Shared/ProviderConfiguration.cs

```diff
@@ -77,6 +77,13 @@ private ProviderConfiguration(NameValueCollection config)
             // Get connection host, port and password.
             // host, port, accessKey and ssl are firest fetched from appSettings if not found there than taken from web.config
             ConnectionString = GetStringSettings(config, "connectionString", null);
+
+            if (!string.IsNullOrEmpty(ConnectionString) &&
```

Comment:
Correct me if I am wrong. After this change user will provide key name for connection string as "connectionString" attribute and there will be connection string with that name in ConfigurationManager.ConnectionStrings section. Please add some unit test to make it clear.